dom.bs

```diff
@@ -1419,6 +1419,107 @@ that gave folks all the wrong ideas. <a>Events</a> do not represent or cause act
+
+<div class=note>
+ For example, a hypothetical <code>doAmazingness({ ... })</code> method could accept a
+ {{CancellationSignal}} object in order to support cancellation as follows:
+
+ <pre class=lang-javascript>
+  const controller = new CancellationController();
+  const signal = controller.signal;
+  doAmazingness({ ..., signal })
+    .then(result => ...)
+    .catch(e => ...);
+
+  ...
+
+  controller.cancel();
+  </pre>
+
+ APIs that require more granular control could extend both {{CancellationController}} and
+ {{CancellationSignal}} according to their needs.
+</div>
+
+<h3 id=interface-cancellationcontroller>Interface {{CancellationController}}</h3>
+
+<pre class="idl">
+[Constructor(), Exposed=(Window,Worker)]
+interface CancellationController {
+  readonly attribute CancellationSignal signal;
+
+  void cancel();
+};
+</pre>
+<dl class=domintro>
+ <dt><code><var>controller</var> = new <a constructor lt="CancellationController()">CancellationController</a>()</code>
+ <dd>Returns a new <var>controller</var> whose {{CancellationController/signal}} is set to a newly
+ created {{CancellationSignal}} object.
+
+ <dt><code><var>controller</var> . </code>{{CancellationController/signal}}
+ <dd>Returns the {{CancellationSignal}} object associated with this object.
+
+ <dt><code><var>controller</var> . </code>{{CancellationController/cancel()}}
+ <dd>Invoking this method will set this object's {{CancellationSignal}}'s
+ [=CancellationSignal/cancelled flag=], thereby signaling to any observers that the associated
+ activity should be cancelled.
+</dl>
+
+The <dfn attribute for=CancellationController><code>signal</code></dfn> attribute must return the value
+to which it was initialized. When a {{CancellationController}} is created, the attribute must be
+initialized to a newly created {{CancellationSignal}} object.
+
+The <dfn method for=CancellationController><code>cancel()</code></dfn> method, when invoked, must run
+these steps:
+
+<ol>
+ <li>Let <var>signal</var> be this object's {{CancellationController/signal}}.
+ <li>Set <var>signal</var>'s [=CancellationSignal/cancelled flag=].
+ <li>Run <var>signal</var>'s [=CancellationSignal/cancellation steps=].
+ <li>[=Fire an event=] named <code>cancel</code> at <var>signal</var>.
+</ol>
+
+<h4 id=interface-cancellationsignal>Interface {{CancellationSignal}}</h4>
+
+<pre class="idl">
+[Exposed=(Window,Worker)]
+interface CancellationSignal : EventTarget {
+  readonly attribute boolean cancelled;
+
+  attribute EventHandler oncancel;
+};
+</pre>
+<dl class=domintro>
+ <dt><code><var>signal</var> . </code>{{CancellationSignal/cancelled}}
+ <dd>Returns true if this {{CancellationSignal}} has been cancelled, and false otherwise.
+</dl>
+
+Each {{CancellationSignal}} has an <dfn for=CancellationSignal>cancelled flag</dfn> which is
+unset unless otherwise specified.
+
+Each {{CancellationSignal}} has an <dfn for=CancellationSignal>cancellation steps</dfn> algorithm
```

Comment:
This might need to be passed in through the creation of the CancelattionController. We'll see what works in prose once FetchController is specified using this.